include/xrpl/basics/Number.h

```diff
@@ -150,6 +150,22 @@ class Number
         return (mantissa_ < 0) ? -1 : (mantissa_ ? 1 : 0);
     }
 
+    Number
+    truncate() const noexcept
+    {
+        if (exponent_ >= 0 || mantissa_ == 0)
+            return *this;
+
+        Number ret = *this;
+        while (ret.exponent_ < 0 && ret.mantissa_ != 0)
+        {
+            ret.exponent_ += 1;
+            ret.mantissa_ /= rep(10);
+        }
+        ret.normalize();
+        return ret;
+    }
```

Comment:
what's the reason for needing to truncate the shares? Wouldn't it work without it?